File: 03/src/main/java/ru/otus/spring/hw3/LocalizedMessageSource.java
package ru.otus.spring.hw3;

import org.springframework.context.MessageSource;
import org.springframework.stereotype.Component;

import java.util.Locale;

@Component
public class LocalizedMessageSource {
    private final Locale locale;
    private final MessageSource messageSource;

    public LocalizedMessageSource(Locale locale, MessageSource messageSource) {
        this.locale = locale;
        this.messageSource = messageSource;
    }

    public String getMessage(String code, Object[] args) {
        return messageSource.getMessage(code, args, locale);
    }
}

File: 04/src/main/kotlin/ru/otus/spring/hw4/LocalizedMessageSource.kt
package ru.otus.spring.hw4

import org.springframework.context.MessageSource
import org.springframework.stereotype.Component
import java.util.*

@Component
class LocalizedMessageSource(
        private val messageSource: MessageSource,
        applicationProperties: ApplicationProperties
) {
    private val locale = Locale(applicationProperties.language)

    fun getMessage(code: String, vararg args: Any?) = messageSource.getMessage(code, args, locale)
}
